src/hera/workflows/script.py

```diff
@@ -558,11 +558,12 @@ def _extract_return_annotation_output(source: Callable) -> List:
         workflow_args = [
             get_args(annotated_type) for annotated_type in annotation_args if get_workflow_annotation(annotated_type)
         ]
+        # If all tuple elements are annotated as Parameter/Artifact
         if len(workflow_args) == len(annotation_args):
             output.extend(workflow_args)
-        elif workflow_args:
+        elif workflow_args:  # Only some tuple elements are annotated as Parameter/Artifact
             raise ValueError(
-                f"Function '{source.__name__}' output has partially annotated return type. Hera allows tuple annotation to be fully annotated or not."
+                f"Function '{source.__name__}' output has partially annotated tuple return type. Tuple elements must be all Annotated as Parameter/Artifact, or contain no Parameter/Artifact annotations for a raw tuple return type."
```

Comment:
nit: could we split this onto two lines? It's got pretty long.
```suggestion
                f"Function '{source.__name__}' output has partially annotated tuple return type."
                " Tuple elements must be all Annotated as Parameter/Artifact, or contain no"
                " Parameter/Artifact annotations for a raw tuple return type."
```

Reply:
Ah, you split it already. I think 3 lines would still be better.

Reply:
Oh thanks! but merged